torch/csrc/autograd/record_function.cpp

```diff
@@ -40,6 +43,22 @@ bool needsInputs() {
   return callback_needs_inputs > 0;
 }
 
+bool isSampledCallbacks() {
```

Comment:
I don't see the need for 4 functions here. Why not:

```
void setSamplingProbability(double prob); {
   if (prob == 1.0) {
     is_sampled_callbacks = false;   
  }
  AT_CHECK(prob > 0 && < 1.0);
  is_sabled_callbacks = true;
  sampling_prob = prob;
}
```

Reply:
I guess if we just always use and pass constant 1.0 then == should work for doubles, although I wouldn't want to explicitly rely on this.. I can simplify interface, but I'd prefer to compare abs(diff) < epsilon